menoh/mkldnn_with_generic_fallback/backend/mkldnn/mkldnn_context.cpp

```diff
@@ -0,0 +1,163 @@
+                                    array arr;
+                                    std::tie(copy_proc, arr) = *found;
+                                    assert(arr.dims().size() == 2 ||
+                                           arr.dims().size() == 4);
+                                    new_copy_procedure_list.push_back(
+                                      copy_proc);
+                                    input_list.push_back(arr);
+                                    auto format =
+                                      arr.dims().size() == 2
+                                        ? mkldnn::memory::format::nc
+                                        : mkldnn::memory::format::nchw;
+                                    is_found_from_other_context = true;
+                                    break;
+                                }
+                            }
+                            assert(is_found_from_other_context);
+                        } while(false);
+                    }
+
+                    // make primitives and
+                    std::vector<mkldnn::primitive> new_primitive_list;
+                    std::vector<std::pair<std::string, mkldnn::memory>>
+                      new_output_memory_list;
+                    std::vector<mkldnn::memory> new_temp_memory_list;
+                    try {
+                        auto factory =
+                          procedure_factory_table_.at(node.op_type);
+                        std::tie(new_primitive_list, new_output_memory_list,
+                                 new_temp_memory_list) =
+                          factory.operator()(current_index, node_list,
+                                             input_list, required_output_table,
+                                             engine_);
+                    } catch(...) { break; }
+                    primitive_list.insert(primitive_list.end(),
+                                          new_primitive_list.begin(),
+                                          new_primitive_list.end());
+                    // add copy procedures
+                    procedure_list.insert(
+                      procedure_list.end(),
+                      std::make_move_iterator(new_copy_procedure_list.begin()),
+                      std::make_move_iterator(new_copy_procedure_list.end()));
+
+                    // update context
+                    variable_memory_table_.insert(
+                      std::make_move_iterator(new_output_memory_list.begin()),
+                      std::make_move_iterator(new_output_memory_list.end()));
+                    temp_memory_list_.insert(
+                      temp_memory_list_.end(),
+                      std::make_move_iterator(new_temp_memory_list.begin()),
+                      std::make_move_iterator(new_temp_memory_list.end()));
+                }
+
+                // when any nodes are not processed
```

Comment:
"when no nodes were processed" might be better.

Reply:
I'll fix. Thanks.